jetty-util/src/main/java/org/eclipse/jetty/util/thread/QueuedThreadPool.java

```diff
@@ -651,105 +684,95 @@ public String toString()
+                            int size = _threadsStarted.get();
+                            if (size > _minThreads)
                             {
-                                // maybe we should shrink?
-                                final int size = _threadsStarted.get();
-                                if (size > _minThreads)
+                                long last = _lastShrink.get();
+                                long now = System.nanoTime();
+                                if (last == 0 || (now - last) > TimeUnit.MILLISECONDS.toNanos(_idleTimeout))
                                 {
-                                    long last = _lastShrink.get();
-                                    long now = System.nanoTime();
-                                    if (last == 0 || (now - last) > TimeUnit.MILLISECONDS.toNanos(_idleTimeout))
-                                    {
-                                        if (_lastShrink.compareAndSet(last, now) && _threadsStarted.compareAndSet(size, size - 1))
-                                        {
-                                            shrink = true;
-                                            break loop;
-                                        }
-                                    }
+                                    if (_lastShrink.compareAndSet(last, now))
+                                        break loop;
                                 }
-                                job = idleJobPoll();
                             }
+
+                            job = _jobs.poll(_idleTimeout, TimeUnit.MILLISECONDS);
                         }
                     }
-                    finally
+
+                    // run job
+                    if (job != null)
                     {
-                        if (_threadsIdle.decrementAndGet() == 0)
+                        if (idle)
                         {
-                            startThreads(1);
+                            idle = false;
+                            if (_threadsIdle.decrementAndGet() == 0)
+                                startThreads(1);
                         }
+
+                        if (LOG.isDebugEnabled())
+                            LOG.debug("run {}", job);
+                        runJob(job);
+                        if (LOG.isDebugEnabled())
+                            LOG.debug("ran {}", job);
+
+                        // Clear interrupted status
+                        Thread.interrupted();
                     }
+
+                    if (!isRunning())
+                        break loop;
+
+                    job = _jobs.poll();
                 }
             }
             catch (InterruptedException e)
             {
-                ignore = true;
                 LOG.ignore(e);
             }
             catch (Throwable e)
             {
+                LOG.warn("Unexpected thread death: {} in {}", this, QueuedThreadPool.this);
                 LOG.warn(e);
```

Comment:
Merge these two `warn()` into one.

Reply:
how would I merge these two lines but keep the stacktrace being logged, seems like it only prints the `Throwable.toString()` if i merge these